symengine/printer.h

```diff
@@ -61,11 +65,27 @@ class Precedence : public BaseVisitor<Precedence>
         bvisit_upoly(x);
     }
 
+    void bvisit(const URatPoly &x)
```

Comment:
Instead of repeating this everywhere, you can do,
```
template <Container, Poly>
bvisit(const UPolyBase<Container, Poly> &x)
{
    bvisit_upoly(static_cast<const Poly &>(x));
}
```